lucetc/src/compiler.rs

```diff
@@ -47,6 +47,91 @@ impl OptLevel {
     }
 }
 
+pub struct CompilerBuilder {
+    pub target: Triple,
```

Comment:
I built this struct before seeing the `with_` conventions in `LucetOpts`.  There is one place (`Lucetc.shared_object_file`) that wants access to this.  Making it `pub` was just a quick way to get something out for review.  Can adopt a different naming convention here to differentiate "accessors" vs. "setters".

Reply:
* How about using `pub(crate)` instead of `pub`? I don't think exposing this to `Lucetc.shared_object_file`, but that would keep `target` from being exposed outside of the crate as well.
* You already noticed the `with_` conventions, following these here would be nice :slightly_smiling_face: I'll tag one of the methods below with an example of the signature change I would hope to see.